tests/fields/test_custom.py

```diff
@@ -0,0 +1,23 @@
+import strawberry
+from strawberry.field import StrawberryField
+
+
+strawberry.field
+
+
+def test_override_get_result():
+    class WorldField(StrawberryField):
+        def get_result(self, *args, **kwargs):
+            res = super().get_result(args, **kwargs)
+            return res + " world"
+
+    def resolve_hello(self) -> str:
+        return "hello"
+
+    @strawberry.type
+    class Query:
+        hello: str = WorldField.evolve_with_resolver(resolve_hello)
```

Comment:
I'm not sure `evolve_with_resolver` is easy to understand, could this follow the same api as `strawberry.field`, so:

`WorldField(resolver=resolve_hello)`?

Reply:
I'll have a try